### DISCUSSION TITLE
Questions about Use of PgModeler 0.9.3.

### DISCUSSION THREAD

**Discussion by PowerUser1234** [pgmodeler/pgmodeler#1620] [General]
-Ask a silly question, but how do I get the snippets editor to come up?

-And whereabouts can I access the "source code input field"?

**Comment by rkhaotix** [comment]
> Ask a silly question, but how do I get the snippets editor to come up?

The snippets editor is located at configuration form, tab Snippets. 

https://www.pgmodeler.io/support/docs/4-introduction?v=0.9.4
https://www.pgmodeler.io/support/docs/45-snippets-settings?v=0.9.4



> And whereabouts can I access the "source code input field"?

You can find source code fields in several parts of the tool. For instance, in Function editor, in tab Definition there's a field called Source code.

Another source code input field is located at SQL Tool where you write commands to send to the server.

https://www.pgmodeler.io/support/docs/6112-sql-execution-panel?v=0.9.4

**Comment by PowerUser1234** [comment]
I am struggling to gather with what you have replied to me with.

-I understand what the source code input field is: [https://www.pgmodeler.io/support/docs/6112-sql-execution-panel?v=0.9.4](https://www.pgmodeler.io/support/docs/6112-sql-execution-panel?v=0.9.4).  What do I click from start to go to it?

-In relation to the source code input field, how do I access the Snippets area to gain code templates for the source code input field?
 Is it possible to actively have both of them on screen at the same time? [https://www.pgmodeler.io/support/docs/45-snippets-settings?v=0.9.3](https://www.pgmodeler.io/support/docs/45-snippets-settings?v=0.9.3)

**Comment by rkhaotix** [comment]
> I am struggling to gather with what you have replied to me with.

Sorry if I'm not making you understand. I'm Brazilian and my English is not very sharp! :)
Let's try again...

> -I understand what the source code input field is: https://www.pgmodeler.io/support/docs/6112-sql-execution-panel?v=0.9.4. What do I click from start to go to it?

From the welcome view (the one that appears right after loading the tool) click the "Manage" button at the top left portion of the main window. 

![image](https://user-images.githubusercontent.com/2205476/147852173-c56b4013-bc2a-4b54-8ea8-b63d2ae94c34.png)

After that, you'll be presented to the SQL Tool. Choose a connection [previously configured](https://www.pgmodeler.io/support/docs/44-connections-settings?v=0.9.4) to start browsing the databases in a server. Once you select the connection and the database one instance of the SQL execution pane will be automatically displayed for the currently browsed database:

![image](https://user-images.githubusercontent.com/2205476/147852251-3723e084-771e-40ef-893b-bbd2d7ec352c.png)

The area pointed by the red arrow is where you can start writing commands to send to the server. Here you can have access to the code snippets in two different ways:

1) From the button "Snippets" at the top of the SQL execution pane. Just navigate through the subitems and select the desired snippet so the code related to it will be displayed in the source input field.

![image](https://user-images.githubusercontent.com/2205476/147852363-16a3723f-ae7e-40d6-9a6e-633befa6a316.png)

2) By bringing the [code completion widget](https://www.pgmodeler.io/support/docs/24-code-completion-widget?v=0.9.4) using ```Ctrl + Space``` in the input field. Note that by using the code completion the snippets are arranged in alphabetical order and not categorized like in item (1). Here, you can select the desired snippet and hit enter so the code related to it is displayed in the input field.

![image](https://user-images.githubusercontent.com/2205476/147852451-59612aef-7066-4596-9371-a10c2d1c63a4.png)

> Is it possible to actively have both of them on-screen at the same time?

No. You can't have the snippets configuration form open while writing commands in an input field.
